### DISCUSSION TITLE
How can I add sourcefile/headerfile to target via rule?

### DISCUSSION THREAD

**Discussion by xq114** [xmake-io/xmake#1685] [Q&A]
现在的rule例子都是直接加到`target:objectfiles()`然后调用编译器，但如果rule有生成headerfile的话这种方法就会有问题（必须等所有rule生成headerfile完毕后才能成功执行编译，但rule中没有栅栏同步功能）

有没有办法让rule生成的文件直接加载到target源文件列表里面，然后target再解析所有源文件编译链接？

**Comment by waruqi** [comment]
on_load或者 on_config 里面生成 走target:add("files",)添加试试
